### PR TITLE
Feature/webumenia 774 add vsg to the list of galleries

### PR DESCRIPTION
**Pull request by igor-kamil**
# Description

Add VSG to the list of galleries and allow download of public-domain artworks

Fixes [WEBUMENIA-774](https://jira.sng.sk/browse/WEBUMENIA-774)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [x] I have requested at least 1 reviewer for this PR
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: public/js/components/map-with-galleries.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  });
+
+
+  $("svg#map path.cls-4").hover(function() {
+    var classes = $( this ).attr('class').split(/\s+/);
+    $.each( classes, function( index, class_name ){
+        // if class_name is uppercased (e.g. SNG, OGD ...) it's gallery ID
+        if (class_name === class_name.toUpperCase()) {
+            console.log(class_name);
+            $(".galleries a#"+ class_name ).addClass('active');
```
Comment (phivk) [inline public/js/components/map-with-galleries.js]: since this is a regular HTML element, you can use `.toggleClass()`

Comment (phivk) [reply public/js/components/map-with-galleries.js]: also, Id prefer to use data attributes where we can, as they are more expressive and clear about what they are used for. So I'd propose to store `gallery_id` in `data-gallery` attributes for the `.galleries a` elements.

see my comment in `informacie.blade.php`

File: resources/views/informacie.blade.php [hunk 2]
```diff
@@ -14,68 +14,28 @@
...
+            <div class="col-md-8 text-center">
+                {!! file_get_contents(public_path('images/mapa.svg')) !!}
             </div>
         </div>
+        <div class="row bottom-space galleries">
+            @foreach(array_chunk($galleries, ceil(count($galleries)/3)) as $chunk)
+                <div class="col-md-4">
+                    <ul class="list-unstyled lead">
+                    @foreach($chunk as $gallery)
+                        <li><a href="{!! URL::to($gallery['url']) !!}" id="{{ $gallery['id'] }}">{{ utrans($gallery['lang_string']) }}</a></li>
```
Comment (phivk) [inline resources/views/informacie.blade.php]: I'd prefer to store the gallery value in a data-attr:
```
<li><a href="{!! URL::to($gallery['url']) !!}" data-gallery="{{ $gallery['id'] }}">{{ utrans($gallery['lang_string']) }}</a></li>
```

File: public/css/less/custom.less [hunk 3]
```diff
@@ -509,14 +509,31 @@ table a:not(.btn):not(.readmore-js-toggle):not(.no-border), .attributes a:not(.b
...
+        transition: fill 0.3s;
+        -webkit-transition-timing-function: cubic-bezier(0.2,1,0.3,1);
+        transition-timing-function: cubic-bezier(0.2,1,0.3,1);
+
+        &:hover, &.active {
+          fill: @primary-color;
+        }
+
+      }
+    }
```
[the remark this answers is not in the document]
Comment (phivk) [reply public/css/less/custom.less]: to consider re UI: if we want the user to create a link between the hovered icon and the active underlined link, wouldn't it be better is icon and active link underline have the same color? 🤔

File: app/Http/routes.php [hunk 4]
```diff
@@ -297,7 +297,68 @@ function()
     Route::get('informacie', function () {
         $items = Item::random(20, ['gallery' => 'Slovenská národná galéria, SNG']);
 
-        return view('informacie', ['items' => $items]);
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply app/Http/routes.php]: thanx :) 
in the future there should be standalone model representing galleries (probably also with db table) - but I guess that's outside of the scope of this task

File: public/js/components/map-with-galleries.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+$("document").ready(function() {
+  $(".galleries a").hover(function() {
+    var gallery_id = $( this ).attr('id');
+    var old_classes = $("svg#map path."+gallery_id ).attr('class');
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply public/js/components/map-with-galleries.js]: thank you for pointing out the `classList()`. however working with classes on svg paths is little bit [tricky](https://stackoverflow.com/questions/36130050/add-class-on-svg-path-element) and I chose this - oldschool but backward compatible approach (working on IE as well)

File: resources/views/informacie.blade.php (hunk 2 above)
Comment (igor-kamil) [reply resources/views/informacie.blade.php]: we use the gallery acronym as unique identifier everywhere around (CEDVU / webumenia / ...)
but in other cases data attribute is definitely right choice! 

right here we store gallery code for querying dom elements. that's what `classes` and `ids` are ideal for. data- attributes are better for storing additional (reusable) informations - but not for querying the element. to read more: https://intuio.at/blog/dont-use-data-attributes-to-find-html-elements-with-js/

File: public/css/less/style.less [hunk 6]
```diff
@@ -21,4 +21,5 @@
 @import "color-list.less";
 @import "article-thumbnail.less";
 @import "artwork-carousel.less";
-@import "zoom-viewer.less";
\ No newline at end of file
+@import "zoom-viewer.less";
+@import "map-with-galleries.less";
```
Comment (phivk) [inline public/css/less/style.less]: @igor-kamil please stick to [component naming conventions](https://wiki.cedvu.sk/display/ODaZS/Component+Library+master+doc), so the name should be:

"TheGalleryMap"

Capitalisation and hyphenation of the actual file names (.less, .blade.php and .js) are currently not very consistent, but let's stick to what is already in place, so I propose:
`the-gallery-map.less`
`the-gallery-map.js`

**Pull request merged**